src/Swarm/Game/Step.hs

```diff
@@ -1399,6 +1400,16 @@ execConst c vs s k = do
             robotMap . at (target ^. robotID) . traverse . systemRobot .= True
 
             ourID <- use @Robot robotID
+            let salvageItems =
+                  if system || creative
+                    then []
+                    else concatMap (uncurry replicateCount) (E.elems salvageInventory)
+                numItems = length salvageItems
+                replicateCount n e =
+                  e ^. entityName & case n of
+                    E.Integer x -> replicate (fromIntegral x)
+                    E.PosInfinity -> replicate 42
+                    E.NegInfinity -> replicate (-42)
```

Comment:
Ooops, this is a problem. How should we salvage infinite inventory? :thinking:

Reply:
On IRC we agreed that it should take 1 tick. The idea is that an infinite item is like an item generator so we move it like it had count 1.

There is still an issue with how to do that. I think some "game state update" on the stack of the reprogrammed salvaged robot is the cleanest solution. An internal `giveInfinite` function could also make sense.